### PR TITLE
ite/it8xxx2: add support for jtag debug interface

### PR DESCRIPTION
**Pull request by Dino-Li**
If enabled, jtag pins will be configured as debug interface at chip startup.

### CODE REVIEW

File: soc/riscv/ite_ec/it8xxx2/Kconfig.soc [hunk 1]
```diff
@@ -118,6 +118,17 @@ config SOC_IT8XXX2_EC_BUS_24MHZ
 	  The clock_frequency of ite,it8xxx2-i2c node (i2c0, i2c1, and i2c2) will
 	  be fixed at 400KHz.
 
+config SOC_IT8XXX2_JTAG_DEBUG_INTERFACE
```
Comment (nordicjm) [inline soc/riscv/ite_ec/it8xxx2/Kconfig.soc:121]: Needs converting to hwmv2

**Review by nordicjm — DISMISSED** [review]

File: soc/riscv/ite_ec/it8xxx2/Kconfig.soc (hunk 1 above)
Comment (Dino-Li) [reply soc/riscv/ite_ec/it8xxx2/Kconfig.soc]: Conflicts have been resolved.

File: soc/ite/ec/common/vector.S [hunk 2]
```diff
@@ -26,6 +26,31 @@ SECTION_FUNC(vectors, __start)
 
 	.option norvc;
 
+#ifdef CONFIG_SOC_IT8XXX2_JTAG_DEBUG_INTERFACE
+	/* Enable JTAG debug interface */
+	la t0, IT8XXX2_GCTRL_PMER3
+	lb t1, 0(t0)
+	ori t1, t1, 0x3
```
Comment (keith-zephyr) [inline soc/ite/ec/common/vector.S:33]: Non-blocking comment: can you define macros for the PMER3 bits you're setting here?

**Review by keith-zephyr — DISMISSED** [review]

File: soc/ite/ec/common/vector.S (hunk 2 above)
Comment (Dino-Li) [reply soc/ite/ec/common/vector.S]: Done.

**Comment by Dino-Li** [comment]
@keith-zephyr Please revisit @GTLin08 Please review

**Review by GTLin08 — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

**Pull request merged**